tools/lint-pr-commit-message.sh

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
+
+# Shell script to lint the message of the first commit in a pull request.
+#
+# Depends on curl, git, node, npm and npx being in $PATH.
+#
+# The pull request is either:
+# 1) supplied as an argument to this shell script
+# 2) TRAVIS_PULL_REQUEST if set and != "false"
+# 3) derived from the HEAD commit via the GitHub API
+
+GH_API_URL="https://api.github.com"
```

Comment:
Remaking my comment from 

On Jenkins we do:
```bash
PR_BASE_BRANCH=curl -k https://api.github.com/repos/${TARGET_GITHUB_ORG}/${TARGET_REPO_NAME}/pulls/${PR_ID} | python -c 'import json,sys;obj=json.load(sys.stdin);print obj["base"]["ref"]'`
```

We also have
https://github.com/nodejs/node/blob/74ba48294b115364b92bbd5c87c8025cd109e303/tools/doc/apilinks.js#L38-L46

Refs: https://stackoverflow.com/questions/3161204/find-the-parent-branch-of-a-git-branch

IMHO The "parent branch deduction" should have it's own `target` (i.e. `case "$1" in ...`) in this script, or their own script

Reply:
This script doesn't care about parent branches though. It asks GitHub for the commits for a PR:
```
curl -s https://api.github.com/repos/nodejs/node/pulls/${PULL_REQUEST_NUMBER}/commits
```
and lints the first commit in the results. 

If the PR isn't specified (either as an argument to the script or via the TRAVIS_PULL_REQUEST env var) then the script asks GitHub what pull requests contain the HEAD commit:
```
curl -s https://api.github.com/search/issues?q=sha:${HEAD_COMMIT}+type:pr+repo:nodejs/node
```

Reply:
Gottcha 👍 
(still worth giving it its own target so we could reuse this logic)